devito/dle/blocking_utils.py

```diff
@@ -1,6 +1,6 @@
 import cgen as c
 
-from devito.dimension import IncrDimension
+from devito.functions.dimension import IncrDimension
```

Comment:
here's the rule that should be applied everywhere (to be added to CONTRIBUTING.md):
* if importing from **within** functions, then the imports should be explicit from each file (e.g., `from devito.functions.dimension import ...`)
* if, instead, importing from a different directory (as here), just import from `functions` (i.e., `from devito.functions import ...`)

Reply:
For some yes, but some of these import are not exported types (not in `__all__`) as internal types so need the explicit path

Reply:
actually most of the ones i tried need the absolute import

Reply:
then we can probably extended the various __all__ and make sure they are not exposed to the user API. This is a convention I'd like to follow, otherwise the import list becomes a mess